### PR TITLE
Feat: gatsby source wordpress experimental

### PR DESCRIPTION
**Pull request by TylerBarnes**
This PR replaces #19101. In that PR I had fully replaced `gatsby-source-wordpress` with my new version. This proved to be annoying for updating from master though since active work is being done on `gatsby-source-wordpress` v3 still. So I've moved  my work into `gatsby-source-wordpress-experimental` for now but left the package name as `gatsby-source-wordpress` to do alpha `@next` releases under.

### CODE REVIEW

File: packages/gatsby-source-wordpress-next/package.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
+{
+  "name": "gatsby-source-wordpress",
```
Comment (m-allanson) [inline packages/gatsby-source-wordpress-next/package.json]: Packages in the `packages` directory will get automatically published, so this name should probably match the parent directory name. Otherwise this may get auto-published as a mainline `gatsby-source-wordpress` release.

File: packages/gatsby-source-wordpress-next/package.json [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  "description": "Source data from WPGraphQL in an efficient and scalable way.",
+  "version": "4.0.0--experimental.1",
```
Comment (m-allanson) [inline packages/gatsby-source-wordpress-next/package.json]: Maybe this could be set to `0.0.0` to denote that it's a pre-release?

Comment (m-allanson) [reply packages/gatsby-source-wordpress-next/package.json]: Or `0.0.0--experimental`

File: examples/using-gatsby-source-wordpress-next/package.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
+{
```
Comment (m-allanson) [inline examples/using-gatsby-source-wordpress-next/package.json]: 💜 for including an example site alongside the plugin

File: docs/docs/themes.md [hunk 4]
```diff
@@ -16,6 +16,5 @@ This means that the configuration and functionality isn’t directly written int
 
 ## Other resources
 
+- [Gatsby theme tutorials](/tutorial/theme-tutorials/)
```
Comment (m-allanson) [inline docs/docs/themes.md]: It looks like there are some unrelated docs changes here. Ok to remove them?

**Comment by m-allanson** [comment]
Oops. I hit the wrong button. That should have been 'request changes' instead of 'approve'.

**Comment by m-allanson** [comment]
Is it worth adding a README to the package, stating that it's experimental, use at your own risk? And that it will eventually be deprecated and turned into a major release of `gatsby-source-wordpress`.

**Comment by laurieontech** [comment]
Excited for this! To echo what @m-allanson said, it'd be good to have a README with a few different things.

- how to use the plugin, examples and information about any configuration variables
- a note about it being experimental
- an explanation of why it exists/is different than the current WP plugin

File: docs/docs/themes.md (hunk 4 above)
Comment (TylerBarnes) [reply docs/docs/themes.md]: Oops! I think this is because I made a new branch from master, but this PR is against the `feat/gatsby-source-v4` branch, not against master, and that branch was made about a month ago!

**Comment by TylerBarnes** [comment]
Thanks for having a look @laurieontech and @m-allanson ! I've added a readme with more info. I meant to open this as a draft PR but accidentally opened it as ready for review and it auto tagged you guys as reviewers. To expand on the description above, the work here isn't yet ready for anyone to use (even experimentally), so I've opened this against the `feat/gatsby-source-wordpress-v4` branch instead of master.
My plan was to make this initial monster PR with rough work, have it generally reviewed so I'm not the only one that's familiar with the code generally, and then break up the remaining work into smaller chunks that are more digestible for CR. @m-allanson does that sound like a good way to do it to you?

File: docs/docs/themes.md (hunk 4 above)
Comment (TylerBarnes) [reply docs/docs/themes.md]: Hmm, I'm not sure how to best remove them. I updated the `feat/gatsby-source-wordpress-v4` branch from master so those commits should exist on that branch, but they still show up in this PR for some reason. I'm worried if I remove them, it will eventually remove them from master when `feat/gatsby-source-wordpress-v4` makes it to master.

File: examples/using-gatsby-source-wordpress-next/package.json (hunk 3 above)
Comment (TylerBarnes) [reply examples/using-gatsby-source-wordpress-next/package.json]: Thanks :) been using it to develop the plugin

File: packages/gatsby-source-wordpress-next/package.json (hunk 1 above)
Comment (TylerBarnes) [reply packages/gatsby-source-wordpress-next/package.json]: Updated it! Thanks Mike :)

**Comment by TylerBarnes** [comment]
🤔 I guess since it's a separate plugin this could potentially be merged to master actually. Not sure if that makes sense or not.

**Comment by m-allanson** [comment]
Oops I totally missed that this was going in to a different branch. My bad.

> I guess since it's a separate plugin this could potentially be merged to master actually. Not sure if that makes sense or not.

Yeah I think this could work? The process would be:

- while this plugin is under development, it can be published as gatsby-source-wordpress-experimental@0.0.x
- when this plugin is ready it can be deprecated, and all changes copied over to `gatsby-source-wordpress`, which will be released as version 4

This avoids having a long running branch and having to merge changes from `master` back into it. What do @gatsbyjs/core think about this approach?

**Comment by m-allanson** [comment]
Oh also for @gatsby/core what are people's thoughts on having [a WordPress plugin at `packages/wp-plugin`](https://github.com/gatsbyjs/gatsby/pull/19948/files#diff-7ece613075823d61405e83c7ef0f1a30)? Should this go elsewhere in the repo to avoid it being published to npm?

**Comment by laurieontech** [comment]
I'll leave the process up to you. While it's a work in progress no need for us to review. Just keep those above notes in mind when you're getting ready to release and polishing up the README!

**Comment by TylerBarnes** [comment]
Thanks for your input @laurieontech and @m-allanson !
@pieh and I talked this over via zoom today and I'm thinking the easiest approach for now is to keep this on a separate branch and publish the individual experimental package using `npm publish` instead of lerna, at least until this makes it to master. 

We're going to keep the `wp-gatsby` WP plugin in `packages` but add `private: true` to the package.json so that we'll still get a changelog generated for that package but it wont be published to npm.
